scrape/target.go

```diff
@@ -346,8 +355,12 @@ func populateLabels(lset labels.Labels, cfg *config.ScrapeConfig) (res, orig lab
 		}
 	}
 
-	preRelabelLabels := lb.Labels()
-	lset = relabel.Process(preRelabelLabels, cfg.RelabelConfigs...)
+	return processLabels(lb.Labels(), cfg)
+}
+
+// processLabels runs relabeling and checks the output.
+func processLabels(preRelabelLabels labels.Labels, cfg *config.ScrapeConfig) (res, orig labels.Labels, err error) {
```

Comment:
I split off the relabeling part of `populateLabels` into a new function because on reload we only need that part.